### PR TITLE
Feature Winchester remake

### PR DESCRIPTION
**Pull request by Enfield-303**
The Winchester has long since been needing an update of its own, as it is one of the very few weapons that still has archaic animations as well as decade old code and asset files, the Winchester 1897 has now had a full overhaul, with new animations, meshes and assets that have been added too it that should hopefully bring it forward to the current DH standard

Features added
- 1907 pattern bayonet, added as the "trench shotgun" was known for having the ability to sport a large bayonet.
- Completely overhauled animations, including new reload, bayonet and firing animations.
- New historically accurate shells, the m19 brass shotgun shell which was used instead of plastic or paper shells as it was more reliable.
- Entirely new character animations, so the gun doesn't just float in the guys hands in game.
- Other minor fixes to the overall makeup of the gun, should hopefully feel a lot better to use now.

Credits: 
m1907 bayonet and texture by `Emmercom camper`
m19 brass shell by `Emmercom camper`
original winchester model and texture by the `29th ID`
Large portion of weapon component code thanks to `Colin basnett`

### CODE REVIEW

File: DH_Weapons/Classes/DH_Winchester1897Attachment.uc [hunk 1]
```diff
@@ -22,100 +21,136 @@ simulated function AnimEnd(int Channel)
...
+    WA_BayonetProneReloadEmpty="idle_bayo_1897"
+    WA_BayonetPreReload="idle_bayo_1897"
+    WA_BayonetPostReload="reload_end_bayo_1897"
+    WA_BayonetAttach="stand_bayattach_1897" 
+    WA_BayonetDetach="stand_bayremove_1897" 
+    WA_BayonetAttachProne="prone_bayattach_1897" 
+    WA_BayonetDetachProne="prone_bayremove_1897"   
+    WA_BayonetWorkBolt="pump_standing_bayo_1897"
+
+    //need to add the pumping anims
```
Comment (cmbasnett) [inline DH_Weapons/Classes/DH_Winchester1897Attachment.uc:62]: Fix the comment here

File: DH_Weapons/Classes/DH_Winchester1897Attachment.uc [hunk 2]
```diff
@@ -22,100 +21,136 @@ simulated function AnimEnd(int Channel)
...
+    //prone iron anims
+
+    PA_ProneIronAnims(0)="prone_slowcrawlF_1897"
+	PA_ProneIronAnims(1)="prone_slowcrawlB_1897"
+	PA_ProneIronAnims(2)="prone_slowcrawlL_1897"
+	PA_ProneIronAnims(3)="prone_slowcrawlR_1897"
+	PA_ProneIronAnims(4)="prone_slowcrawlL_1897"
+	PA_ProneIronAnims(5)="prone_slowcrawlR_1897"
+	PA_ProneIronAnims(6)="prone_slowcrawlB_1897"
+	PA_ProneIronAnims(7)="prone_slowcrawlB_1897"
```
Comment (cmbasnett) [inline DH_Weapons/Classes/DH_Winchester1897Attachment.uc:141]: Inconsistent indentation

File: DH_Weapons/Classes/DH_Winchester1897Fire.uc [hunk 3]
```diff
@@ -26,8 +27,13 @@ defaultproperties
...
-    ShellIronSightOffset=(X=9.5,Y=1.6,Z=-2.4)
+    ShellHipOffset=(X=0,Y=0,Z=0)
+    ShellIronSightOffset=(X=0,Y=0,Z=0)
     FireForce="AssaultRifleFire"
+    MuzzleBone="muzzle"
+    ShellEmitBone="ejector"
     ShakeOffsetMag=(X=3.0,Y=2.0,Z=8.0)
+
+    
+
```
Comment (cmbasnett) [inline DH_Weapons/Classes/DH_Winchester1897Fire.uc:38]: Unnecessary whitespace

File: DH_Engine/Classes/DHWeaponRegistry.uc [hunk 4]
```diff
@@ -169,4 +169,5 @@ defaultproperties
     Records(109)=(ClasSName="DH_Weapons.DH_DT29Weapon",WeaponNames=("dt29","dt29"))
     Records(110)=(ClasSName="DH_Weapons.DH_MKB42HWeapon",WeaponNames=("mkb42","mkb42h"))
     Records(111)=(ClassName="DH_Weapons.DH_FG42ScopedWeapon",WeaponNames=("fg42scoped", "zfg42"))
+    Records(112)=(ClassName="DH_Weapons.DH_Winchester1897Weapon",WeaponNames=("winchester", "shotgun", "1897"))
```
Comment (cmbasnett) [inline DH_Engine/Classes/DHWeaponRegistry.uc:172]: Unnecessary duplication (sorry!)

File: DH_Engine/Classes/DHProjectileWeapon.uc [hunk 5]
```diff
@@ -153,10 +153,15 @@ var     bool            bCanUseIronsights;      // allows firing from a shoulder
 
 enum EWeaponComponentAnimationDriverType
 {
+    DRIVER_None,                // Not driven by anything other than animation notifies.
```
Comment (cmbasnett) [inline DH_Engine/Classes/DHProjectileWeapon.uc:156]: This can actually be removed, it was a WIP thing that I left in by mistake

**Review by cmbasnett — CHANGES_REQUESTED** [review]

File: DH_Weapons/Classes/DH_Winchester1897Attachment.uc (hunk 1 above)
Comment (Enfield-303) [reply DH_Weapons/Classes/DH_Winchester1897Attachment.uc:62]: Resolved in latest commit

File: DH_Weapons/Classes/DH_Winchester1897Fire.uc (hunk 3 above)
Comment (Enfield-303) [reply DH_Weapons/Classes/DH_Winchester1897Fire.uc:38]: Resolved in latest commit

File: DH_Engine/Classes/DHWeaponRegistry.uc (hunk 4 above)
Comment (Enfield-303) [reply DH_Engine/Classes/DHWeaponRegistry.uc:172]: Resolved in latest commit

File: DH_Engine/Classes/DHProjectileWeapon.uc (hunk 5 above)
Comment (Enfield-303) [reply DH_Engine/Classes/DHProjectileWeapon.uc:156]: Resolved in latest commit

**Review by cmbasnett — APPROVED** [review]
